test/sanity/code-smell/update-bundled.py

```diff
@@ -0,0 +1,104 @@
+# (c) 2018, Ansible Project
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+import glob
+import json
+import os.path
+import sys
+from distutils.version import LooseVersion
+
+import packaging.specifiers
+
+from ansible.module_utils.urls import open_url
+
+
+def compat_libs(basedir):
+    for filename in glob.glob(os.path.join(basedir, 'lib/ansible/compat/*/__init__.py')):
+        if 'compat/tests' in filename:
+            # compat/tests doesn't bundle any code
+            continue
+        yield filename
+
+    yield os.path.join(basedir, 'lib/ansible/module_utils/distro/__init__.py')
+    yield os.path.join(basedir, 'lib/ansible/module_utils/six/__init__.py')
+    yield os.path.join(basedir, 'lib/ansible/module_utils/compat/ipaddress.py')
+    # backports.ssl_match_hostname should be moved to its own file
+    yield os.path.join(basedir, 'lib/ansible/module_utils/urls.py')
+
+
+def get_bundled_metadata(filename):
+    with open(filename, 'r') as module:
+        for line in module:
+            if line.strip().startswith('_BUNDLED_METADATA'):
+                data = line[line.index('{'):].strip()
+                break
+        else:
+            raise Exception('{0} contained no metadata.  Could not check for updates'.format(filename))
+        metadata = json.loads(data)
+    return metadata
+
+
+def get_latest_applicable_version(pypi_data, constraints=None):
+    latest_version = "0"
+    if 'version_constraints' in metadata:
+        version_specification = packaging.specifiers.SpecifierSet(metadata['version_constraints'])
+        for version in pypi_data['releases']:
+            if version in version_specification:
+                if LooseVersion(version) > LooseVersion(latest_version):
+                    latest_version = version
+    else:
+        latest_version = pypi_data['info']['version']
+
+    return latest_version
+
+
+if __name__ == '__main__':
+    needs_attention = False
+    basedir = os.path.join(os.path.dirname(__file__), '../../..')
+
+    for filename in compat_libs(basedir):
+        filename = os.path.normpath(filename)
+        try:
+            metadata = get_bundled_metadata(filename)
+        except Exception as e:
+            needs_attention = True
+            print('WARNING: {0}'.format(e))
```

Comment:
Warnings are not supported. This will result in an error. Print the error in the format: `path: message`

Reply:
Yeah, I do want it to result in an error for the test harness.  I'm not fussy about whether the message should read warning or error.

Reply:
Changed the message to read ERROR instead of WARNING.